src/compiler.rs

```diff
@@ -137,9 +138,18 @@ impl AmberCompiler {
         })
     }
 
-    pub fn import_std() -> String {
-        [
-            include_str!("std/main.ab")
-        ].join("\n")
+    pub fn resolve_std(path: String, meta: &mut ParserMetadata, token_import: Option<Token>) -> Result<String, Failure> {
+        let mut std_modules = HashMap::new();
+
+        std_modules.insert("".to_string(), include_str!("std/main.ab").to_string());
+        std_modules.insert("std".to_string(), include_str!("std/main.ab").to_string());
+        std_modules.insert("strings".to_string(), include_str!("std/strings.ab").to_string());
+        std_modules.insert("fs".to_string(), include_str!("std/fs.ab").to_string());
```

Comment:
Is there any problem with this namespace definition?
I think it will affect future language design.

For example, I feel `std/string` is better than `std/strings`.

Reply:
I'm also struggling to understand why just the `"std"` isn't good enough. Is there any deep reason behind this since if we just import functions separately and they got tree-shaken, @b1ek?

Reply:
> I'm also struggling to understand why just the `"std"` isn't good enough

as of right now, its fine as it is but when we'd want to grow it to have an imagick, curl, fs and other kind of modules it will be a mess if they are placed in one file. i have actually started working on this PR because i wanted to make a curl module but didn't want to place it in `main.rs`

i mean, you dont do stuff like `use std::{read_file_to_string, SocketAddr, IoError}` in rust, right? even if unused stuff will get tree shaken away

Reply:
I see. Why not import everything publicly from the `main.ab` and have `string.ab` and other modules?

```
pub import * from "./string.ab"
```

Oh... this could be hard to implement for the path resolution to work. I see now.

Reply:
Yeah, this would require the installer to fetch the stdlib as well however this would make this issue https://github.com/Ph0enixKM/Amber/issues/170 impossible to implement. I think that for now we could leave the current state of the PR:

```js
import * from "std/string"
```

If we leave it like so - it's easy to 

I think that we should also discuss how the paths should work with package manager in the future. Here are some ideas I had in mind:

Not sure if we want to get our hands dirty and maintain package registry. This would require some funding and good sotware architecture as it'd be easy to abuse it. If we decide to keep things simple - we could offer importing packages directly from git repositories as the main way to fetch packages.

```js
// Pull the code from GitHub repository (latest)
import * from "gh:b1ek/amber-library"
// Pull the code from GitHub repository with version
import * from "gh:b1ek/amber-library@2.1.3"
// Pull the code from url (the version should be dependent on the url)
import * from "https://example.com/amber-library"
// Import the code from path
import * from "./path/to/file"
```

This would require us to implement an `amber.lock` file to keep the track on versioning of all dependencies.

Reply:
> this would require the installer to fetch the stdlib

why? the stdlib exists only at compile time. there's no need to source it in the script at the runtime.

as for package registry, that is probably the job for a third party cli application, as its been done for node, rust, and many other languages

Reply:
I wanted to point out that (Deno)[https://deno.land/x/manual@v1.20.2/linking_to_external_code.md?source=] did it in the early stage but now I see that they discourage it as people can easily put malware in the third party url.

Never mind. This was not a good idea.

I think you have a good approach here.